### ISSUE TITLE
New Kubernetes service connection causes an error Could not find any secrets associated with the Service Account.

### ISSUE THREAD

**Issue by michelefa1988** [microsoft/terraform-provider-azuredevops#670]
I see the starting that clusters created with AKS version 1.24+, AzureDevops will no longer support the following code due to a change made by Kubernetes.  (For reference see comments  [here](https://developercommunity.visualstudio.com/t/New-Kubernetes-service-connection-causes/10138123):

My goal is that after creating an AKS cluster in terraform, I also want to create AKS Service Endpoint of type "AzureSubscription." This seems to not work anymore due to the error "Could not find any secrets associated with this service account"

```
resource "azuredevops_serviceendpoint_kubernetes" "serviceendpoint_kubernetes" {
  project_id            = data.azuredevops_project.p.id
  service_endpoint_name = var.service_endpoint_name
  apiserver_url         = var.apiserver_url
  authorization_type    = "AzureSubscription"

  azure_subscription {
    subscription_id   = data.azurerm_subscription.current.subscription_id
    subscription_name = var.subscription_name
    tenant_id         = data.azurerm_subscription.current.tenant_id
    resourcegroup_id  = var.resourcegroup_id
    namespace         = "default"
    cluster_name      = var.aks_cluster_name
    cluster_admin     = var.cluster_admin
  }
```

What is the long term solution for this? Is Microsoft working on a fix? When is the fix expected? I would also like to keep this issue open until is is resolved so that the community  can keep track on progress.

**Comment by xuzhang3** [comment]
@michelefa1988  https://developercommunity.visualstudio.com/t/New-Kubernetes-service-connection-causes/10138123  suggest create the service connection with type `KubeConfig`,  have tried that solution?

**Comment by michelefa1988** [comment]
@xuzhang3  Yes, but they are all workarounds and not a permanent solution. 

I'm expecting Microsoft to come up with a more permanent fix eventually?

**Comment by fgrossin** [comment]
I got the same problem with 1.24+
As soon as I downgrade to 1.23.12, the k8s service connection is correctly created.

I am not really expecting a fix from Microsoft since the aforementioned solution (https://developercommunity.visualstudio.com/t/New-Kubernetes-service-connection-causes/10138123) is classified as "Closed - Not a bug" and there is a workaround (KubeConfig or service account)

I'll work on updating my code to use one of these other solutions
